_rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

```diff
@@ -0,0 +1,431 @@
+---
+id: fd3a94
+name: Links with identical accessible names and context serve equivalent purpose
+rule_type: atomic
+description: |
+  This rule checks that links with identical accessible names and context resolve to the same or equivalent resources.
+accessibility_requirements:
+  wcag20:2.4.4: # Link Purpose (In Context) (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Anne Thyme Nørregaard
+    - Carlos Duarte
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'attr-lowercase'
+  - 'alt-require'
+---
+
+## Applicability
+
+This rule applies to any set of two or more HTML or SVG elements which
+
+- have the [semantic role][] of `link`, or a role that inherits from the `link` role; and
+- are in the same [web page (HTML)][]; and
+- are [included in the accessibility tree][included in the accessibility tree]; and
+- have [matching][] [accessible names][accessible name] that are not empty (`""`); and
+- have the same [programmatically determined link context][].
+
+**Note:** The test target for this rule is the full set of link elements that share the same [matching][] [accessible name][] and [programmatically determined link context][].
+
+## Expectation
+
+When followed, the links in each set of target elements resolve to the [same resource][] or to [equivalent resources](#equivalent-resource).
+
+**Note**: Resolving the links includes potential redirects, if the redirects happen instantly.
+
+## Assumptions
+
+- This rule assumes that the purpose of the links with identical [accessible names][accessible name] and [context][programmatically determined link context] would not be ambiguous to users in general, which is the exception mentioned in [Success Criterion 2.4.4 Link Purpose (In Context)][sc244]. If the links are ambiguous to users in general, users of assistive technologies are not at a disadvantage when viewing the links, which makes it more of a general user experience concern than an accessibility issue.
+- This rule assumes that, within the context of the test subject, the description provided by the [accessible name][] of a link can only accurately describe one resource (notably, homonyms alone are not used as link names). Thus, if two or more links have the same [accessible name][] but resolve to different resources, at least one of them does not describe its purpose.
+- This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.
```

Comment:
Language is an input. We don't need to assume we can determine it, if we can't, input is missing and the rule can't be executed. This can be taken out I think.

Reply:
The CG decision was to keep this assumption nonetheless:
https://www.w3.org/2020/01/09-act-r-minutes.html#item04